packages/web-components/src/progress-bar/progress-bar.ts

```diff
@@ -36,9 +40,96 @@ export class ProgressBar extends BaseProgress {
   /**
    * The validation state of the progress bar
    * @public
-   * @remarks
    * HTML Attribute: validation-state
    */
   @attr({ attribute: 'validation-state' })
   public validationState: ProgressBarValidationState | null = null;
+
+  /**
+   * The value of the progress
+   * @internal
+   * HTML Attribute: value
+   */
+  @attr({ converter: nullableNumberConverter })
+  public value?: number | null;
+  protected valueChanged(): void {
+    this.updateAriaValueNow();
+    this.updatePercentComplete();
+  }
+
+  /**
+   * The minimum value
+   * @internal
+   * HTML Attribute: min
+   */
+  @attr({ converter: nullableNumberConverter })
+  public min?: number;
+  protected minChanged(): void {
+    if (this.$fastController.isConnected) {
+      this.updateAriaValueMin();
+      this.updatePercentComplete();
+    }
+  }
+
+  /**
+   * The maximum value
+   * @internal
+   * HTML Attribute: max
+   */
+  @attr({ converter: nullableNumberConverter })
+  public max?: number;
+  protected maxChanged(): void {
+    if (this.$fastController.isConnected) {
+      this.updateAriaValueMax();
+      this.updatePercentComplete();
+    }
+  }
+
+  /**
+   * Indicates progress in %
+   * @internal
+   */
+  @observable
+  public percentComplete: number = 0;
+
+  /**
+   * @internal
+   */
+  public constructor() {
+    super();
+
+    this.elementInternals.role = 'progressbar';
+  }
+
+  /**
+   * @internal
+   */
+  public connectedCallback(): void {
+    super.connectedCallback();
+    this.updateAriaValueNow();
+    this.updateAriaValueMin();
+    this.updateAriaValueMax();
+    this.updatePercentComplete();
```

Comment:
Are these calls necessary in the `connectedCallback()` considering that the Observable `-Changed` methods already call each of these respective methods? Would moving the default assignments from `updatePercentComplete()` to the initial declarations cause these to get called regardless?

```ts
@attr({ converter: nullableNumberConverter })
public max!: number = 100;

@attr({ converter: nullableNumberConverter })
public min!: number = 0;

@attr({ converter: nullableNumberConverter })
public value!: number = 0;
```

Reply:
I've removed these - I don't think we want or need to default here. On one hand, max gets automatically set to 100 on the ARIA side of things and min to 0. Value we really only want reflected _if_ there is a value provided, otherwise we'll use 0 to calculate the percent complete.

Reply:
@radium-v I had to add this check back in because the changed() functions are not being called on connection at least within playwright. Let me know if you have thoughts on this.